src/SIM/SIMoutput.C

```diff
@@ -1016,8 +1016,8 @@ bool SIMoutput::writeGlvN (const Matrix& norms, int iStep, int& nBlock,
   NormBase* norm = myProblem->getNormIntegrand(mySol);
 
   Matrix field;
-  std::array<IntVec,30> sID;
-  const size_t maxN = sID.size();
+  std::vector<IntVec> sID;
```

Comment:
A more elegant solution would be just
`std::vector<IntVec> sID(norms.rows());`
which always will be big enough since `norms.rows()` will be equal to `field.rows()` if you see what extractElmRes does. Then most of the other changes (except deleting `maxN`) can be removed.